rotkehlchen/exchanges/binance.py

```diff
@@ -1400,33 +1400,46 @@ def _deserialize_fiat_movement(
                 error=msg,
             )
         else:
-            return AssetMovement(
+            events = [AssetMovement(
                 location=self.location,
-                category=category,
-                address=address,
-                transaction_id=tx_id,
-                timestamp=timestamp,
+                event_type=event_type,
+                timestamp=ts_sec_to_ms(timestamp),
                 asset=asset,
-                amount=amount,
-                fee_asset=asset,
-                fee=fee,
-                link=link_str,
-            )
+                balance=Balance(amount),
+                unique_id=tx_id,
+                extra_data=maybe_set_transaction_extra_data(
+                    address=address,
+                    transaction_id=tx_id,
+                ),
+            )]
+            if fee != ZERO:
+                events.append(AssetMovement(
+                    event_identifier=events[0].event_identifier,
+                    location=self.location,
+                    event_type=event_type,
+                    timestamp=ts_sec_to_ms(timestamp),
+                    asset=asset,
+                    balance=Balance(fee),
+                    is_fee=True,
+                ))
+
+            return events
 
         return None
 
-    def _deserialize_asset_movement(self, raw_data: dict[str, Any]) -> AssetMovement | None:
+    def _deserialize_asset_movement(self, raw_data: dict[str, Any]) -> list[AssetMovement] | None:
         """Processes a single deposit/withdrawal from binance and deserializes it
 
         Can log error/warning and return None if something went wrong at deserialization
         """
         try:
+            event_type: Literal[HistoryEventType.DEPOSIT, HistoryEventType.WITHDRAWAL]
             if 'insertTime' in raw_data:
-                category = AssetMovementCategory.DEPOSIT
+                event_type = HistoryEventType.DEPOSIT
                 timestamp = deserialize_timestamp_from_intms(raw_data['insertTime'])
                 fee = Fee(ZERO)
             else:
-                category = AssetMovementCategory.WITHDRAWAL
+                event_type = HistoryEventType.WITHDRAWAL
                 timestamp = deserialize_timestamp_from_date(
                     date=raw_data['applyTime'],
```

Comment:
you can do `ts_sec_to_ms` here and avoid doing it both in the movement and the fee